pkg/controller/core/dynamicresourceallocationconfig_controller.go

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package core
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+
+	kueuealpha "sigs.k8s.io/kueue/apis/kueue/v1alpha1"
+	"sigs.k8s.io/kueue/pkg/cache"
+	"sigs.k8s.io/kueue/pkg/features"
+)
+
+// +kubebuilder:rbac:groups=kueue.x-k8s.io,resources=dynamicresourceallocationconfigs,verbs=get;list;watch
+
+// DynamicResourceAllocationConfigReconciler keeps the cache's DeviceClass→logical-resource
+// mapping in sync with the singleton DynamicResourceAllocationConfig object.
+// It is only registered when the DynamicResourceAllocation feature gate is enabled.
+type DynamicResourceAllocationConfigReconciler struct {
+	client    client.Client
+	cache     *cache.Cache
+	log       logr.Logger
+	namespace string
+}
+
+func NewDRAConfigReconciler(client client.Client, c *cache.Cache, namespace string) *DynamicResourceAllocationConfigReconciler {
+	return &DynamicResourceAllocationConfigReconciler{
+		client:    client,
+		cache:     c,
+		log:       ctrl.Log.WithName("dra-config-reconciler"),
+		namespace: namespace,
+	}
+}
+
+func (r *DynamicResourceAllocationConfigReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	if !features.Enabled(features.DynamicResourceAllocation) {
```

Comment:
https://github.com/kubernetes-sigs/kueue/pull/5873/files#diff-a59d3d6d0bd5d684e293231de0698d9af2df82e7373cfb898dd8b0d89f8ffe87R85

You are checking the feature gate before you even start the controller. I don't think you need the check for the feature gate here.

Reply:
Indeed, it is better to skip registering the controller

Reply:
This feature gate check here captures the request here: https://github.com/kubernetes-sigs/kueue/pull/5873/files#r2223069702

Should we just have the feature gate at the start of the controller and remove everywhere else? That would mean not feature gate check in cache.go? @mimowo looking for guidance here.